llama.h

```diff
@@ -258,7 +259,7 @@ extern "C" {
         bool logits_all;  // the llama_decode() call computes all logits, not just the last one (DEPRECATED - set llama_batch.logits instead)
         bool embedding;   // embedding mode only
         bool offload_kqv; // whether to offload the KQV ops (including the KV cache) to GPU
-        bool do_pooling;  // whether to pool (sum) embedding results by sequence id (ignored if no pooling layer)
+        enum llama_pooling_type pooling_type;  // whether to pool (sum) embedding results by sequence id (ignored if no pooling layer)
```

Comment:
Move this next to `rope_scaling_type` above and pass as `int32_t`

Reply:
Should I have it be an `int32_t` in `llama_cparams` and `llm_build_context` as well? Otherwise will need a `static_cast<llama_pooling_type>` in `llama_new_context_with_model`.